CBinding/CMake/CMakeProject.cs

```diff
@@ -261,17 +265,31 @@ void AddFile (FilePath fileName, string targetName)
 
 				FileService.CreateDirectory (file.ParentDirectory.Combine (outputDirectory));
 
+	//			Stream generationResult;
+				DefaultCCompiler = PropertyService.Get<string> ("CBinding.DefaultCCompiler", new GccCompiler ().Name);
 				monitor.BeginStep ("Generating build files.");
-				Stream generationResult = ExecuteCommand ("cmake", "../", outputDirectory, monitor);
-				results = ParseGenerationResult (generationResult, monitor);
+				/*	if (guiOptions.default_c_compiler.Name == "msvc") {
+						generationResult = ExecuteCommand ("cmake", "../ -G \"Visual Studio 15 2017\"", outputDirectory, monitor);
+						buildFlag = 1;
+					} else {
+						generationResult = ExecuteCommand ("cmake", "../ -G \"MinGW Makefiles\"", outputDirectory, monitor); //Default is gcc.
+					}
+	*/        //		results = ParseGenerationResult (generationResult, monitor);
+				string projectName = string.Format ("{0}.{1}", fileFormat.ProjectName, "sln");
+				results = await cmakeToolchain.GenerateMakefiles (projectName, outputDirectory, monitor, DefaultCCompiler);
```

Comment:
We can use this line after removing async , this builds successfully but it is not calling GenerateMakefiles method?  
results = cmakeToolchain.GenerateMakefiles (projectName, outputDirectory, monitor, DefaultCCompiler).Result;